contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,177 @@
+
+2. **Angular uses TypeScript.** Angular applications are built using the TypeScript language, a superset of JavaScript. TypeScript provides <tooltip content="Static typing involves type checking during compile-time as opposed to run-time. Types are associated with variables not values.">static typing</tooltip>. This helps the compiler show warnings about any potential errors in the code, before the app runs. Another advantage is that TypeScript provides code completion using <tooltip content="IntelliSense provides active hints as code is added">IntelliSense.</tooltip>
+
+    Below is an example of static typing using TypeScript.
+    
+    ```typescript
+    // TypeScript
+    let name: string; // <-- static typed!
+    ```
+    
+    ```javascript
+    // JavaScript (ES6)
+    let name;
+    ```
+    <box type="info"> TypeScript was created and is maintained by Microsoft. It has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> over the last few years making it reliable to use.</box>
+                                                  
+3. **Angular is backed by Google.** Google announced Long-Term Support (LTS) for Angular back in 2017. This means Google plans to further scale up the Angular ecosystem.
+
+### What does an Angular app look like?
+
+Let’s dive into some code, with a hello world example. The code is applicable for versions Angular 2 and above.
+
+```html
+<!doctype html>
+<html>
+  <head>
+    <!-- The various &lt;script&gt; tags needed to load Angular -->
+    <!-- For a full list, see https://angular.io/docs/ts/latest/quickstart.html -->
+  </head>
+  <body>
+    <my-app></my-app>
+  </body>
+</html>
+```
+
+The one unique thing in the above example is the `<my-app>` element. That is not an element you would normally use in a web-based application. Angular extends the vocabulary of HTML by allowing you do define your own tags.
+These custom tags are known as components.
+
+Let's now look at some TypeScript code that defines the `<my-app>` component.
+
+```typescript
+import { Component } from "@angular/core";
+
+@Component({
+  selector: "my-app",
+  template: `
+    <h1>Hello World</h1>
+  `
+})
+export class AppComponent {}
+```
+
+In Angular we use the `@Component` tag to mark a class as a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: my-app` is how Angular knows what to do when it sees a `<my-app>` tag in HTML.
+
+* The `template` property controls what HTML gets rendered when this component is loaded.
+
+A visual to summarize the above :
+
+![Component](https://angular.io/generated/images/guide/architecture/component-databinding.png)
+
+## Why Angular?
+
+Angular is a popular frontend tool and provides developers with advantages when compared to other frameworks/libraries. Listed below are a few benefits of using Angular.
```

Comment:
The first sentence doesn't add much value but throws in a vague term `frontend tool` (earlier you called Angular a `JavaScript framework`). Perhaps get rid of that sentence?